service/collector/client.go

```diff
@@ -0,0 +1,47 @@
+package collector
+
+import (
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/service/credential"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// kubernetesCredentialNamespace is the namespace in which we store credentials.
+	kubernetesCredentialNamespace = "giantswarm"
+
+	// kubernetesLabelSelector is the label selector we use to retrieve credentials.
+	kubernetesLabelSelector = "app=credentiald"
+)
+
+// getUniqueClientSets fetches all unique Azure clients.
+func getUniqueClientSets(k8sClient kubernetes.Interface, environmentName string) ([]*client.AzureClientSet, error) {
+	credentialList, err := k8sClient.CoreV1().Secrets(kubernetesCredentialNamespace).List(metav1.ListOptions{
```

Comment:
The idea here is to look at the credentials we have, as opposed to the azure configs, and use the credentials directly to create clients.

For example, in the case where a resource group exists, but we don't have any CRs that reference the account in which the resource group exists, we can still be alerted.
We also avoid having duplicate metric errors due to watching resources - we just can list the secrets on each collect.

We should probably actually check that we don't have multiple credentials referencing the same account, but I thought i'd do that in a follow up.

Reply:
This might be a stupid question, but then there's something I don't know:

> For example, in the case where a resource group exists, but we don't have any CRs that reference the account in which the resource group exists, we can still be alerted.

What CRs do you mean in here? I thought this would mean `AzureConfig`, but as I understand this code, it's required 10 lines below in order to be able to create `clientSet`. Something that I'm missing or just don't know/understand correctly?

Reply:
I do mean `AzureConfig`.

`credential.GetAzureConfig` returns a `*client.AzureClientSetConfig`, not an AzureConfig CR.

Reply:
> We should probably actually check that we don't have multiple credentials referencing the same account, but I thought i'd do that in a follow up.

Isn't it fine if two orgas has credentials to the same account? Do we require to have unique roles for each secret?